crates/cairo-lint-core/tests/test_files/manual_unwrap_or_default/manual_unwrap_or_default

```diff
@@ -0,0 +1,574 @@
+
+//! > cairo_code
+fn main() {
+  let a: Option<felt252> = Option::Some(1);
+
+  match a {
+    Option::Some(v) => v,
+    Option::None => Default::default()
+  };
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: This can be done in one call with `.unwrap_or_default()`
+ --> lib.cairo:6:3
+  |
+6 |     match a {
+  |  ___-
+7 | |     Option::Some(v) => v,
+8 | |     Option::None => Default::default()
+9 | |   };
+  | |___-
+  |
+
+//! > fixed
+fn main() {
+  let a: Option<felt252> = Option::Some(1);
+
+  a.unwrap_or_default();
+}
+
+//! > ==========================================================================
+
+//! > manual unwrap or default for match with empty string
+
+//! > cairo_code
+fn main() {
+  let x: Option<ByteArray> = Option::Some("Hello");
+  match x {
+    Option::Some(v) => v,
+    Option::None => ""
+  };
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: This can be done in one call with `.unwrap_or_default()`
+ --> lib.cairo:4:3
+  |
+4 |     match x {
+  |  ___-
+5 | |     Option::Some(v) => v,
+6 | |     Option::None => ""
+7 | |   };
+  | |___-
+  |
+
+//! > fixed
+fn main() {
+  let x: Option<ByteArray> = Option::Some("Hello");
+  x.unwrap_or_default();
+}
+
+//! > ==========================================================================
+
+//! > manual unwrap or default for match with fixed array
+
+//! > cairo_code
+fn main() {
+  let x: Option<[u64; 5]> = Option::Some([1, 2, 3, 4, 5]);
+  match x {
+    Option::Some(v) => v,
+    Option::None => [0; 5]
+  };
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: This can be done in one call with `.unwrap_or_default()`
+ --> lib.cairo:4:3
+  |
+4 |     match x {
+  |  ___-
+5 | |     Option::Some(v) => v,
+6 | |     Option::None => [0; 5]
+7 | |   };
+  | |___-
+  |
+
+//! > fixed
+fn main() {
+  let x: Option<[u64; 5]> = Option::Some([1, 2, 3, 4, 5]);
+  x.unwrap_or_default();
+}
+
+//! > ==========================================================================
+
+//! > manual unwrap or default for match with new
+
+//! > cairo_code
+fn main() {
+  let x: Option<Array<u128>> = Option::Some(array![1, 2, 3, 4, 5]);
+  match x {
+    Option::Some(v) => v,
+    Option::None => ArrayTrait::new()
+  };
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: This can be done in one call with `.unwrap_or_default()`
+ --> lib.cairo:4:3
+  |
+4 |     match x {
+  |  ___-
+5 | |     Option::Some(v) => v,
+6 | |     Option::None => ArrayTrait::new()
+7 | |   };
+  | |___-
+  |
+
+//! > fixed
+fn main() {
+  let x: Option<Array<u128>> = Option::Some(array![1, 2, 3, 4, 5]);
+  x.unwrap_or_default();
+}
+
+//! > ==========================================================================
+
+//! > manual unwrap or default for match with tuple
+
+//! > cairo_code
+fn main() {
+  let x: Option<(ByteArray, u128, bool)> =Option::Some(("James", 90, true));
+  match x {
+    Option::Some(v) => v,
+    Option::None => ("", 0, false)
+  };
+}
```

Comment:
can you try with the tuple but one of the elements isn't the default value ?

Reply:
Okay

Reply:
https://github.com/JoE11-y/cairo-lint/blob/main/crates/cairo-lint-core/tests/test_files/manual/manual_unwrap_or_default#L549